### PR TITLE
Improve routing of PrestaShop application

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Routing current organization has already been improved by @sarjon, let's go deeper for 1.7.5. 
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | All tests should pass as they don't rely on hardcoded URLs, but some of them have been changed to better respect the menu organization (mostly for the first migrated pages). As we use the router to generate URLs and not hardcoded values, this is not a break of API! Also, URLs from API part have not been changed (if we consider them as part of a REST API...).

> Want an overview of application routing? => try php bin/console debug:router

admin
api

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9137)

### CODE REVIEW

File: admin-dev/themes/new-theme/js/app/pages/stock/router/index.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default new VueRouter({
   mode: 'history',
   base: (() => {
     const hasIndex = /(index\.php)/.exec(window.location.href);
-    return `${window.data.baseUrl}${hasIndex ? '/index.php' : ''}/stock`;
+    return `${window.data.baseUrl}${hasIndex ? '/index.php' : ''}/sell/stocks`;
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/js/app/pages/stock/router/index.js:36]: note that I'm not proud of my fix neither to the current code, but fixing it "the right way" should be done in a specific pull request.

**Comment by mickaelandrieu** [comment]
ping @fatmaBouchekoua can you explain to me how to reproduce the failing test, I can create a product with a price, a reference a description and 2 pictures manually, so I don't really know what bug I have introduced here :disappointed:

**Comment by PierreRambaud** [comment]
@mickaelandrieu This selector `#growls-default > .growl-notice > .growl-message:not(:empty)` isn't found, it can be changed by `.growl-message:not(:empty)` and the test pass

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    resource: "modules/modules.yml"
+    prefix: /modules/
+
+#_payment:
+#    resource: "payment/*.yml"
+#    prefix: /payment/
+#
+#_shipping:
+#    resource: "shipping/*.yml"
+#    prefix: /shipping/
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml]: Missing eof

**Comment by mickaelandrieu** [comment]
> This selector #growls-default > .growl-notice > .growl-message:not(:empty) isn't found, it can be changed by .growl-message:not(:empty) and the test pass

Maybe... but why it worked before and not now? I'm not comfortable having to change the tests to make them pass :(

**Comment by PierreRambaud** [comment]
Yeah, it's weird, I try this selector directly in chrome extension and it returns nothing :/

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml (hunk 2 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml]: fixed, ty

**Comment by PierreRambaud** [comment]
A little update and merge please :D

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml (hunk 2 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml]: Hum, still missing but don't care

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/design/_design.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+_theme_catalog:
+    resource: "theme_catalog.yml"
+    prefix: /themes-catalog/
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/config/routing/admin/improve/design/_design.yml]: Missing new line

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/design/theme_catalog.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 admin_theme_catalog:
-    path: /addons-store
+    path: /
     methods: [GET]
     defaults:
         _controller: 'PrestaShopBundle:Admin\Improve\Design\ThemeCatalog:index'
-        _legacy_controller: AdminThemesCatalog
+        _legacy_controller: AdminThemesCatalog
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/config/routing/admin/improve/design/theme_catalog.yml]: Missing new line

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/design/_design.yml (hunk 3 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/config/routing/admin/improve/design/_design.yml]: :/

missing CS validator configuration and build in travis.

This is the last time I fix a coding style issue, I don't have time for this /c  @eternoendless

**Comment by mickaelandrieu** [comment]
@eternoendless LGTM? the more we wait, the bigger this pull request will be ^^

File: src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml (hunk 2 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/config/routing/admin/improve/_improve.yml]: fixed

**Comment by PierreRambaud** [comment]
:+1:

**Comment by mickaelandrieu** [comment]
At this point I dunno why e2e are broken and I can't reproduce it :/

**Comment by mickaelandrieu** [comment]
> @mickaelandrieu This selector #growls-default > .growl-notice > .growl-message:not(:empty) isn't found, it can be changed by .growl-message:not(:empty) and the test pass

this is still the same issue :)

**Comment by PierreRambaud** [comment]
@mickaelandrieu No prob mate, I will have a check on tomorrow if it's ok for you =)

**Comment by Quetzacoalt91** [comment]
We need a rebase before merge, some other PRs have been included here.

**Comment by mickaelandrieu** [comment]
@PierreRambaud My HERO <3

**Comment by PierreRambaud** [comment]
I just update assets after pull upstream develop branch, I don't understand why there is #9198 because it has already been merged ><

**Comment by PierreRambaud** [comment]
haha, I use rebase commands, that's why there are some commits related to develop branch :)

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Thank you @mickaelandrieu & @PierreRambaud

**Pull request merged**